comp110_architecture.tex

```diff
@@ -5,41 +5,150 @@
+\newline
+\newline
+\begin{figure}[h]
+	\centering
+	\includegraphics[width=0.7\linewidth]{Fig6}
+	\caption{ - Example of the level after the algorithm has been applied to \textit{Infinite Mario Bros.}}.
+	\label{Fig6}
+	\centering
+	\includegraphics[width=0.7\linewidth]{Fig7}
+	\caption{ -Example of the level after the algorithm has been applied to \textit{XRick}}.
+	\label{Fig7}
+\end{figure}\cite{fausto}
+
+
+\subsection{Genetic Algorithm (GA)}
+
+Genetic Algorithm (GA) is a search heuristic that will mimic the natural selection process, a taxonomy of a evolutionary algorithm that runs through hundreds of evolutions to determine the most optimized method for the task at hand\cite{tax}. Using this algorithm, Infispel, an adaptation of the famous platform game Spelunky is a good example where this is used. A Genetic Algorithm (GA) along with graphs of the rooms and the paths connecting them is used to populate the levels with items and objects. While a agent-based Digger algorithm is used for the rooms structure depending on the difficulty selected Fig.8. Primarily for map building in a first-person shooter games the digger algorithm can provide a platformer game with interesting level builds Fig.8. \cite{walaa}. The way in which the digger creates a level is its placed in one of the 16 rooms and is set of in a random direction, assigning each cell a value of 1(on) or 0(off). The digger will repeat this for a total of 35 moves, walls are then dug where necessary for room connection, then repeated again for the 16 rooms.  
+\newline
+\begin{figure}[h]
+	\centering
+	\includegraphics[width=0.7\linewidth]{Fig8}
+	\caption{ - Three Examples of the digger outcome with increasing difficulty}.
+	\label{Fig8}
+\end{figure}\cite{walaa}
+
+Through this method a developer could create a PCG game based only on these two Algorithms, but they would require quite a bit of modification before implementation unlike the ICA\cite{walaa,fausto}.
+
+\subsection{Endless Web (EW)}
+
+The last and biggest method would be using the Launchpad along with a PCG algorithm to create a game that continuously builds the game depending on your play style\cite{web}. With EW the approach taken was to use the Launchpad as the primary PCG source then extensively modify it to suit their needs. Fig.9 shows the capabilities of the end product.
+
+\begin{figure}[h]
+	\centering
+	\includegraphics[width=0.7\linewidth]{Fig9}
+	\caption{ - Three different scenarios that can come from Endless Web}.
+	\label{Fig9}
+\end{figure}\cite{web}
+
+The game is dependant on the way you tune the world around you, by using different \textit(Tuning Portals) and the proximity to friendly NPC's the world around you will automatically generate and change\cite{web}. The development of such a PCG game would require a great deal of time and cost, for a small indie company it would be impractical to use such a method to create a game.
 
 \section{Recommendation}
```

Comment:
There is no need for a separate section called 'recommendation'. Fold this into the closing of your main body and the conclusion as appropriate.